src/Aspire.Hosting.Azure.AppContainers/ContainerAppContext.cs

```diff
@@ -488,7 +488,7 @@ BicepValue<string> GetHostValue(string? prefix = null, string? suffix = null)
             EndpointProperty.Host or EndpointProperty.IPV4Host => GetHostValue(),
             EndpointProperty.Port => port.ToString(CultureInfo.InvariantCulture),
             EndpointProperty.HostAndPort => GetHostValue(suffix: $":{port}"),
-            EndpointProperty.TargetPort => targetPort is null ? AllocateContainerPortParameter() : targetPort,
+            EndpointProperty.TargetPort => targetPort is null ? AllocateContainerPortParameter() : $"{targetPort}",
```

Comment:
Did you tell the fallback case of "{targetPort}"

Reply:
`ProjectWithManyReferenceTypes` test covers this.